packages/app-bridge/src/types/Document.ts

```diff
@@ -91,23 +101,35 @@ export type DocumentApi = Simplify<
 
 export type Document = CamelCasedPropertiesDeep<DocumentApi>;
 
-export type DocumentLink = Simplify<Document & DocumentAsLink & { linkSettings: { newTab: boolean } }>;
+export type DocumentLink = Simplify<Document & DocumentAsLink>;
 
 export type DocumentLibrary = Simplify<
     Document & DocumentAsNoneLink & { settings?: { project: number }; mode: DocumentLibraryMode }
 >;
 
 export type DocumentStandard = Simplify<Document & DocumentAsNoneLink>;
 
+export type DocumentLinkSettings = CamelCasedPropertiesDeep<DocumentLinkSettingsApi>;
+
+export enum FormLinkStateType {
+    textAndIcon = 'ICON_TEXT',
+    iconOnly = 'ICON',
+    textOnly = 'TEXT',
+}
+
+export enum FormLinkStateIconPosition {
+    right = 'RIGHT',
+    left = 'LEFT',
+}
+
 type DocumentLinkRequest = {
     id: number;
     title: string;
     linkUrl: string;
-    linkSettings: {
-        newTab: boolean;
-    };
+    linkSettings: DocumentLinkSettings;
```

Comment:
You need to take care of Create and Update types as mentioned in Clarify PR.

Reply:
All items on `DocumentLinkSettings` are optional but `newTab`, Create and Update reply on this one, so it should be cover already, isn't it?